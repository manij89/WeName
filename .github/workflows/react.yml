name: React Test
on: [push]
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Install dependencies for client
        run: npm ci --prefix client
      - name: Run tests for client
        run: npm test --prefix client
 