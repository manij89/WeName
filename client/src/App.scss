@import url('https://fonts.googleapis.com/css2?family=Montserrat+Alternates:wght@100;300;400;600&display=swap');

* {
  font-family: 'Montserrat Alternates', sans-serif;
  box-sizing: border-box;

}

