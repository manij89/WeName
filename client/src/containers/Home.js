import React, { useState } from 'react';
import '../styles/home.scss';
import { Wave } from 'react-animated-text';
import stork from '../assets/stork.svg';
import grass from '../assets/grass.svg';

import '../styles/info.scss';
import { Link } from 'react-router-dom';
import { Paper, List, ListItemText, ListItem } from '@material-ui/core';


export default function Home(props) {
  const [inView, setInView] = useState(false);


  return (
    <div>
      <div className={inView ? 'home active' : 'home'}>
        <div className='logo'>
          <Wave
            text='WENAME'
            speed={7}
            effect="verticalFadeIn"
            effectChange={2.5}
            effectDirection='up'
            iterations={1}

          />
        </div>

        <button
          className='home_button'
          onClick={() => setInView(true)}
        >
          Start
      </button>

        <img className='stork' src={stork} alt='stork' />
        <img className='grass' src={grass} alt='grass' />
      </div>

      <div className={inView ? 'info-container active' : 'info-container'}>
        <Paper
          elevation={7}
          className='info-card'
        >
          <h1>Find your favorite baby names</h1>
          <List>
            <div>
              <ListItem>
                <ListItemText
                  primary='Register and go to your "Account"'
                />
              </ListItem>
              <ListItem>
                <ListItemText
                  primary='Share or enter the personal code to play with your partner'
                />
              </ListItem>
              <ListItem>
                <ListItemText
                  primary='Keep swiping and check your matches'
                />
              </ListItem>
            </div>
          </List>
          <div>

            <Link to='/register'>
              <button
                className='info-button'
              >Sign up</button>
            </Link>
            
          </div>
        </Paper>
      </div>
    </div>
  )
}
