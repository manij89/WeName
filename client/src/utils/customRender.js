import React from 'react'
import { render } from '@testing-library/react';
import { Provider } from "react-redux";
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import reducer from '../redux/reducers';
import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
const middleware = [thunk];
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;



 const customStore = createStore(
  reducer,
  composeEnhancers(applyMiddleware(...middleware))
);

 const customRender = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store=customStore,
    ...renderOptions
  } = {}) => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <Router history={history}>{children}</Router>
    </Provider>
  )

  return {
    ...render(ui, {
      wrapper: Wrapper,
      ...renderOptions,
    }), 
    history,
    store,
  };
}

export default customRender;
