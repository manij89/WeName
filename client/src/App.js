import './App.scss';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Route, Switch } from 'react-router-dom';
import Home from './containers/Home';
import Info from './containers/Info';
import Register from './containers/Register';
import Login from './containers/Login'; 
import Game from './containers/Game';
import Profile from './containers/Profile';
import Matches from './components/Matches';
import Deck from './containers/Deck';

import {user} from './redux/actions';



function App() {
  return (

    <Switch>
      <Route
        exact
        path='/'
        component={Home}
      />
      <Route
        exact
        path='/info'
        component={Info}
      />
      <Route
        exact
        path='/register'
        component={Register}
      />
      <Route
        exact
        path='/login'
        component={Login}
      />
      <Route
        exact
        path='/game'
        component={Game}
      />
      <Route
        exact
        path='/game/:gender'
        render={routeProps => (
          <>
            <Deck gender={routeProps.match.params.gender} />
          </>
        )}
      />
      <Route
        exact
        path='/profile'
        component={Profile}
      />
      <Route
        exact
        path='/matches'
        component={Matches}
      />
    </Switch>

  );
}

export default App;
