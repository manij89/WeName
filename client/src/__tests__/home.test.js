import React from 'react';
import { cleanup} from '@testing-library/react'
import { createMemoryHistory } from 'history';
import App from '../App';
import Home from '../containers/Home';


import customRender from '../utils/customRender'

afterEach(cleanup)

it('register snapshot', () => {
  const history = createMemoryHistory();
  history.push('/profile');
  const { asFragment } = customRender(<App />, { history });
  
  expect(asFragment(<Home />)).toMatchSnapshot()
})
