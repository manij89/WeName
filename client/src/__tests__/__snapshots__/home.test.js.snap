// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register snapshot 1`] = `<DocumentFragment />`;
